dask/bytes/tests/test_hdfs.py

```diff
@@ -179,8 +256,52 @@ def test_parquet_pyarrow(hdfs):
                       columns=list('abcd'))
     ddf = dd.from_pandas(df, npartitions=4)
 
-    ddf.to_parquet(hdfs_fn, engine='pyarrow')
-    assert len(hdfs.ls(fn))  # Files are written
+    with dask.set_options(hdfs_driver=driver.name):
+        ddf.to_parquet(hdfs_fn, engine='pyarrow')
+
+    assert len(driver.hdfs.ls(fn))  # Files are written
+
+    with dask.set_options(hdfs_driver=driver.name):
+        ddf2 = dd.read_parquet(hdfs_fn, engine='pyarrow')
 
-    ddf2 = dd.read_parquet(hdfs_fn, engine='pyarrow')
     assert len(ddf2) == 1000  # smoke test on read
+
+
+def test_pyarrow_glob(pa_hdfs):
```

Comment:
Here you show how the globs are different, but we would like them to be the same. Perhaps a test across engines showing that at least simpler globs are the same?

Reply:
I moved the glob implementation out into its own file, and applied it to `hdfs3` as well (which had inconsistent behavior). Modified the test to test both drivers.